controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -239,7 +239,36 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
 
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_SEARCH_MESSAGE";
 
-    auto tlvAlMacAddressType = cmdu_rx.addClass<ieee1905_1::tlvAlMacAddressType>();
+    std::shared_ptr<ieee1905_1::tlvAlMacAddressType> tlvAlMacAddressType     = nullptr;
+    std::shared_ptr<ieee1905_1::tlvSearchedRole> tlvSearchedRole             = nullptr;
+    std::shared_ptr<ieee1905_1::tlvAutoconfigFreqBand> tlvAutoconfigFreqBand = nullptr;
+    std::shared_ptr<wfa_map::tlvSupportedService> tlvSupportedServiceIn      = nullptr;
+    std::shared_ptr<wfa_map::tlvSearchedService> tlvSearchedService          = nullptr;
+    int type = cmdu_rx.getNextTlvType();
```

Comment:
In this case, I think that `switch-case` is more appropriate:
```suggestion
    while ((int type = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
        switch (type) {
            case int(wfa_map::eTlvTypeMap::TLV_SUPPORTED_SERVICE):
                LOG(DEBUG) << "Found TLV_SUPPORTED_SERVICE TLV";
                tlvSupportedServiceIn = cmdu_rx.addClass<wfa_map::tlvSupportedService>();
                break;
            case int(ieee1905_1::eTlvType::TLV_AL_MAC_ADDRESS_TYPE):
                ...
                break;
            default:
                LOG(ERROR) << ...
                return false;
        }
    }
```

Also note that I defined `int type = cmdu_rx.getNextTlvType()` in the while condition, so it is not necessary to do the same outside the `while` loop. Same goes to the assignment at the end of the while.